packages/@o3r/localization/schematics/ng-update/v11-6/index.ts

```diff
@@ -0,0 +1,28 @@
+import type {
+  Rule,
+  Tree,
+} from '@angular-devkit/schematics';
+import {
+  createSchematicWithMetricsIfInstalled,
+  findFilesInTree,
+} from '@o3r/schematics';
+
+const regexp = /translations\s*:\s*(?:Readonly<)?([A-Z][\w]*)(?:>)?\s*=\s*({[^;]+})\s*(as\s*const)?\s*;/g;
```

Comment:
To handle different linter rule guidelines it can be an easy fix to turn the `;` non-mandatory :
```suggestion
const regexp = /translations\s*:\s*(?:Readonly<)?([A-Z][\w]*)(?:>)?\s*=\s*({[^;]+})\s*(as\s*const)?\s*;?/g;
```

Reply:
It's more difficult to detect the end of the object then
`;` is used before in the regexp for that `({[^;]+})`

Reply:
Then I propose:
```
/translations\s*:\s*(?:Readonly<)?([A-Z][\w]*)(?:>)?\s*=\s*({(?:[^:]+:\s*(?:".*"|'.*'|\s*,\s*))*\s*})\s*(as\s*const)?\s*;?/g
```

Reply:
Does not work for complex object like the one in this example 
https://regex101.com/r/2HTPCa/1

Reply:
Indeed but it is working for localisation, which is what you want to parse, don't you?
(And it does support `;` within the translation key even if I am not sure we do support it farer in the extractor 😅)